Makefile

```diff
@@ -30,6 +30,10 @@ format: install-goimports
 	@echo "> Formating code..."
 	@goimports -l -w -local ${PACKAGE_BASE} ${SRC}
 
+lint:
+	@echo "> Linting code..."
+	@golangci-lint run -c golangci.yaml
```

Comment:
We should be able to [automatically install](https://github.com/golangci/golangci-lint#windows-macos-and-linux) this similar to `goimports` and `mockgen`. Is there a reason we wouldn't want to do that?

Reply:
Re-read your description and noticed that you called out that this aspect isn't done yet. 👍